concordium-std/src/prims.rs

```diff
@@ -304,9 +341,20 @@ extern "C" {
         column: u32,
     );
 
-    #[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+    /// Reporting back an error, only exists in debug mode
+    // TODO: Shouldn't this be feature gated with debug? Trivially adding it gives other errors.
```

Comment:
This is feature gated `#[cfg(feature = "wasm-test")]` at the top of the `extern 'C'`

Reply:
Then why is the `debug_print` feature gated?

Reply:
right good thing this turned up, we don't want to change behavior for these, so `report_error` should now have `#[cfg(target_arch = "wasm32")]` to match the behavior from before.

`debug_print` should be feature gated separately from `wasm-test`, so it should be moved out into the above `extern 'C'` with `#[cfg(feature = "debug")]` like it was before. This function might be used within a debug build of a smart contract.